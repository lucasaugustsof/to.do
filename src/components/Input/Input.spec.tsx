import { describe, it, expect } from 'vitest'

import { render, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'

import { Input } from '.'

const testID = 'test:input'
describe('Components/Input', () => {
  it('should render the Form component correctly', () => {
    const { asFragment } = render(<Input />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should write a text correctly', async () => {
    const testInputValue = 'Typed text'

    render(<Input data-testid={testID} />)

    const inputComponent = screen.getByTestId<HTMLInputElement>(testID)
    await userEvent.type(inputComponent, testInputValue)

    expect(inputComponent.value).toBe(testInputValue)
  })

  it('should render correct placeholder', () => {
    render(<Input data-testid={testID} />)

    const inputComponent = screen.getByTestId<HTMLInputElement>(testID)

    expect(inputComponent.placeholder).not.toEqual('Placeholder')
  })

  it('should be disabled when the disable property is activated', () => {
    render(<Input data-testid={testID} disabled />)

    const inputComponent = screen.getByTestId<HTMLInputElement>(testID)

    expect(inputComponent.disabled).toBeTruthy()
  })
})
