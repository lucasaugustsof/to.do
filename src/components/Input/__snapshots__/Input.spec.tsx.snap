// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components/Input > should render the Form component correctly 1`] = `
<DocumentFragment>
  <input
    class="_container_2e074e"
    placeholder="Adicione uma nova tarefa"
    type="text"
  />
</DocumentFragment>
`;
